### DISCUSSION TITLE
Not all QR codes are read, what is wrong?

### DISCUSSION THREAD

**Discussion by ignaciofatou** [journeyapps/zxing-android-embedded#749] [Q&A]
Hi

Hello, in my new project with these liberias I am able to scan all the QR codes perfectly except for the one I need for my project, I have attached an image of the code, I have tried doing it with the original and I still cannot scan it. I have tried other QR code readers from the PlayStore and they read it perfectly without any problem. What is going wrong?

this is the QR image that is failing:
https://postimg.cc/CBn5FnS8

Manifest:
        <uses-sdk tools:overrideLibrary="com.google.zxing.client.android" />
        ......
        <application
        ...
        android:hardwareAccelerated="true"
         ...        
        <activity
            android:name="com.journeyapps.barcodescanner.CaptureActivity"
            android:screenOrientation="fullSensor"
            tools:replace="screenOrientation" />
        </application>

Grandle:
   minSdk 21

    compileOptions {
        coreLibraryDesugaringEnabled true
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
    
    dependenies {
        implementation 'com.journeyapps:zxing-android-embedded:4.3.0'
        coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'
    }
    
ScanActiviy.java:
barcodeLauncher.launch(new ScanOptions());

    private final ActivityResultLauncher<ScanOptions> barcodeLauncher = registerForActivityResult(new ScanContract(),
            result -> {
                if(result.getContents() != null) {
                   ...
                   ...
                }
            });

**Comment by ignaciofatou** [comment]
ScanOptions options = new ScanOptions();
        options.addExtra(Intents.Scan.SCAN_TYPE, Intents.Scan.INVERTED_SCAN);
        barcodeLauncher.launch(options);

    private final ActivityResultLauncher<ScanOptions> barcodeLauncher = registerForActivityResult(new ScanContract(),
            result -> {
                if(result.getContents() == null) {
                    Intent originalIntent = result.getOriginalIntent();
                    if (originalIntent == null) {
                        Toast.makeText(MainActivity.this, "Cancelado", Toast.LENGTH_LONG).show();
                    } else if(originalIntent.hasExtra(Intents.Scan.MISSING_CAMERA_PERMISSION)) {
                        Toast.makeText(MainActivity.this, "Cancelado, debe dar permisos de Cámara a la Aplicación", Toast.LENGTH_LONG).show();
                    }
                }
                //Se Escaneo Correctamente el Codigo QR
                else {
                  ...
                }
            });

**Reply by rkistner** [reply]
Using the `INVERTED_SCAN` like you have here works for me. You need it because the barcode has a white foreground on a dark background, which is non-standard.

You also need a bigger border around the barcode (although I can still scan without it).
